decryptor/postgresql/pg_escape_decryptor.go

```diff
@@ -197,7 +197,7 @@ func (decryptor *PgEscapeDecryptor) readDataLength(reader io.Reader) (uint64, []
 		return 0, decryptor.octLengthBuf[:octLenCount], base.ErrFakeAcraStruct
 	}
 	decryptor.outputSize += octLenCount
-	binary.Read(bytes.NewBuffer(decryptor.lengthBuf[:]), binary.LittleEndian, &length)
+	binary.Read(bytes.NewReader(decryptor.lengthBuf[:]), binary.LittleEndian, &length)
```

Comment:
what's the difference?

Reply:
bytes.Reader wrap []byte with method Read to implement io.Reader and just return data from bytes
bytes.NewBuffer has overhead with internal buffer for short slices and store []byte too. NewReader doesn't have extra internal buffer and uses less memory

Reply:
> uses less memory

👍